### PR TITLE
gh-104016: Fixed off by 1 error in f string tokenizer

### PR DESCRIPTION
**Pull request by jx124**
* Issue: gh-104016

### CODE REVIEW

**Comment by Fidget-Spinner** [comment]
DIsclaimer: This is my friend who I'm helping guide to contribute to CPython. I will thus exclude myself from approving this to prevent any conflict of interest which may arise.

File: Lib/test/test_fstring.py [hunk 1]
```diff
@@ -1548,6 +1548,10 @@ def test_syntax_error_after_debug(self):
                                 "f'{1=}{1;'",
                                 "f'{1=}{1;}'",
                             ])
+                            
+    def test_nested_fstring_max_stack_level(self):
+        with self.assertRaises(SyntaxError):
+            compile('f"{1 1:' + ('{f"1:' * 199), "?", "exec")
```
Comment (jx124) [inline Lib/test/test_fstring.py]:
```suggestion

    def test_nested_fstring_max_stack_level(self):
        with self.assertRaises(SyntaxError):
            compile('f"{1 1:' + ('{f"1:' * 199), "?", "exec")
```

**Comment by pablogsal** [comment]
I was actually preparing a fix for this problem. I think the correct approach here is to separate the two limits. I will update your PR with my approach and we can land it afterwards

**Review by pablogsal — APPROVED** [review]

**Comment by pablogsal** [comment]
Great job @jx124 🚀

**Comment by brettcannon** [comment]
@pablogsal  ☝️ blowing out memory on WASI.

**Comment by pablogsal** [comment]
> @pablogsal point_up blowing out memory on WASI.

On it

**Comment by arhadthedev** [comment]
For the record: the WASM test got fixed gh-104071.

**Pull request merged**